twopaneview/src/TwoPaneView.tsx

```diff
@@ -0,0 +1,241 @@
+import React, {Fragment, Component} from 'react';
+import {
+  SafeAreaView,
+  StyleSheet,
+  ScrollView,
+  View,
+  ViewProps,
+  Text,
+  StatusBar,
+  Dimensions,
+  ViewStyle,
+} from 'react-native';
+import { DualScreenInfo } from 'react-native-dualscreeninfo'
+
+const TPV_Orientation = {
+  Horizontal: 'horizontal',
+  Vertical: 'vertical',
+};
+
+const TPV_PanePriority = {
+  Pane1: 'pane1',
+  Pane2: 'pane2',
+};
+
+const PaneMode = {
+  Auto: 'Auto',
+  Single: 'Single',
+  Dual: 'Dual',
+};
+
+type Props = {
+} & Partial<DefaultProps>;
+
+type DefaultProps = Readonly<typeof defaultProps>;
+
+const defaultProps = { panePriority: 'pane1', panePriorityVerticalSpanning: 'horizontal', 
+  paneMode: 'Auto', twoPaneWidth: 640, onModeChanged: ()=>{}
+}
+
+type State = {
+  dims: any,
+  spanning: boolean, 
+  panePriority?: string,
+  panePriorityVerticalSpanning?: string,
+  paneMode: string,
+  twoPaneWidth: number,
+  onModeChanged: any,
+};
+
+export class TwoPaneView extends Component<Props, State> {
+  state: State = {
+    dims: Dimensions.get('window'),
```

Comment:
>dims: Dimensions.get('window'), [](start = 4, length = 31)

Discussed offline.  There are still a bunch of changes needed, like to stop using Dimensions, which can be done as follow-up changes.  Filed [https://github.com/microsoft/react-native-dualscreen/issues/9](https://github.com/microsoft/react-native-dualscreen/issues/9) to track.

Reply:
Right, eventually we would have not need for Dimensions. However, I am still having trouble getting the logic right with windowRects as exposed in DualScreenInfo